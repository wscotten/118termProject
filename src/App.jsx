import React, { PureComponent } from 'react';
import logo from './styles/logo.svg';
import './styles/App.css';
import Container from './modules/TriangleIncludesTangentLine/Container';

export default class App extends PureComponent {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">118 Term Project: A triangle includes a tangent line.</h1>
        </header>
        <Container />
      </div>
    );
  }
}
