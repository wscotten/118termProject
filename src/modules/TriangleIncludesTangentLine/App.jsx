import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Col } from 'react-bootstrap';
import Form from './components/Form';
import Canvas from './components/Canvas';
import isSolvable from '../../utils/isSolvable';
import {
  getPointA,
  getPointB,
  getPointC,
  getPointD,
  getLineAB,
  getLineAC,
  getLineAD,
  getLineBC,
  getLineBD,
  getLineCD,
  perimeterABD,
  perimeterABC,
  perimeterBCD,
  perimeterCircle,
  areaCircle,
  angleABC,
  angleCAB,
  angleBCA,
  angleABD,
  angleBDA,
  angleBDC,
  angleCBD,
  areaABC,
  smallArcBD,
  largeArcBD,
} from '../../utils/geometryEquations';
import { onChange } from './redux';

class Container extends PureComponent {
  render() {
    const {
      CanvasProps,
      FormProps,
      onChange,
    } = this.props;
    return (
      <div>
        <Col lg={1} xs={4} />
        <Col xs={4} >
          <Canvas {...CanvasProps} />
        </Col>
        <Col lg={7} xs={12} >
          <Form
            {...FormProps}
            onChange={onChange}
          />
        </Col>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  CanvasProps: {
    inputFields: state.inputFields,
    isSolvable: isSolvable(state.inputFields),
  },
  FormProps: {
    inputFields: state.inputFields,
    isSolvable: isSolvable(state.inputFields),
    pointA_X: getPointA()[0],
    pointA_Y: getPointA()[1],
    pointB_X: getPointB(state.inputFields.radius)[0],
    pointB_Y: getPointB(state.inputFields.radius)[1],
    pointC_X: getPointC(state.inputFields)[0],
    pointC_Y: getPointC(state.inputFields)[1],
    pointD_X: getPointD(state.inputFields)[0],
    pointD_Y: getPointD(state.inputFields)[1],
    lineAB: getLineAB(state.inputFields),
    lineAC: getLineAC(state.inputFields),
    lineAD: getLineAD(state.inputFields),
    lineBD: getLineBD(state.inputFields),
    lineBC: getLineBC(state.inputFields),
    lineCD: getLineCD(state.inputFields),
    perimeterABD: perimeterABD(state.inputFields),
    perimeterABC: perimeterABC(state.inputFields),
    perimeterBCD: perimeterBCD(state.inputFields),
    perimeterCircle: perimeterCircle(state.inputFields.radius),
    areaCircle: areaCircle(state.inputFields),
    angleABC: angleABC(),
    angleCAB: angleCAB(state.inputFields),
    angleBCA: angleBCA(state.inputFields),
    angleABD: angleABD(state.inputFields),
    angleBDA: angleBDA(state.inputFields),
    angleBDC: angleBDC(state.inputFields),
    angleCBD: angleCBD(state.inputFields),
    areaABC: areaABC(state.inputFields),
    smallArcBD: smallArcBD(state.inputFields),
    largeArcBD: largeArcBD(state.inputFields),
  },
});

const mapDispatchToProps = dispatch => bindActionCreators({
  onChange,
}, dispatch);

const App = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Container);

export default App;
