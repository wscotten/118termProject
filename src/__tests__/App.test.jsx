import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

it('snapshot test', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
