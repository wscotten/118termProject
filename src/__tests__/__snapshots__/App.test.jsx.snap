// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      118 Term Project: A triangle includes a tangent line.
    </h1>
  </header>
  <Container />
</div>
`;
